aea/helpers/dialogue/base.py

```diff
@@ -132,48 +134,153 @@ def __str__(self):
         )
+
         self._outgoing_messages = []  # type: List[Message]
         self._incoming_messages = []  # type: List[Message]
 
     @property
     def dialogue_label(self) -> DialogueLabel:
-        """Get the dialogue lable."""
+        """
+        Get the dialogue label.
+
+        :return: The dialogue label
+        """
         return self._dialogue_label
 
+    @property
+    def role(self) -> "Role":
+        """
+        Get the agent's role in the dialogue.
+
+        :return: the agent's role
+        """
+        return self._role
+
+    @role.setter
+    def role(self, role: "Role") -> None:
+        """
+        Set the agent's role in the dialogue.
+
+        :param role: the agent's role
+        :return: None
+        """
+        self._role = role
+
     @property
     def is_self_initiated(self) -> bool:
-        """Check whether the agent initiated the dialogue."""
+        """
+        Check whether the agent initiated the dialogue.
+
+        :return: True if the agent initiated the dialogue, False otherwise
+        """
         return self._is_self_initiated
 
     @property
     def last_incoming_message(self) -> Optional[Message]:
-        """Get the last incoming message."""
+        """
+        Get the last incoming message.
+
+        :return: the last incoming message if it exists, None otherwise
+        """
         return self._incoming_messages[-1] if len(self._incoming_messages) > 0 else None
 
     @property
     def last_outgoing_message(self) -> Optional[Message]:
-        """Get the last incoming message."""
+        """
+        Get the last outgoing message.
+
+        :return: the last outgoing message if it exists, None otherwise
+        """
         return self._outgoing_messages[-1] if len(self._outgoing_messages) > 0 else None
 
+    @property
+    def last_message(self) -> Optional[Message]:
+        """
+        Get the last message.
+
+        :return: the last message if it exists, None otherwise
+        """
+        if (
+            self.last_incoming_message is not None
+            and self.last_outgoing_message is not None
+        ):
+            last_incoming_message_id = cast(
+                int, self.last_incoming_message.get("message_id")
+            )
+            last_outgoing_message_id = cast(
+                int, self.last_outgoing_message.get("message_id")
+            )
+            return (
+                self.last_outgoing_message
+                if last_outgoing_message_id > last_incoming_message_id
+                else self.last_incoming_message
+            )
+        elif (
+            self.last_incoming_message is not None
+            and self.last_outgoing_message is None
+        ):
+            last_incoming_message_id = cast(
+                int, self.last_incoming_message.get("message_id")
+            )
+            return self.last_incoming_message if last_incoming_message_id == 1 else None
```

Comment:
That seems wrong...if there is a last incoming message, then this should be returned. The feasibility check (`if last_incoming_message_id == 1`) should always return `True` here as only feasible messages should be permitted into the dialogue

Reply:
Done